e2e/portalicious/pages/RegistrationsPage.ts

```diff
@@ -161,6 +240,185 @@ class RegistrationsPage extends BasePage {
     await this.table.tableRows.nth(row).click({ button: 'right' });
     await this.page.getByLabel(action).click();
   }
+
+  async clickAndSelectExportOption(option: string) {
+    await this.exportButton.click();
+    await this.page.getByRole('menuitem', { name: option }).click();
+  }
+
+  async assertExportButtonIsHidden() {
+    await expect(this.exportButton).toBeHidden();
+  }
+
+  async clickProceedToExport() {
+    await this.proceedButton.click();
+  }
+
+  async exportAndAssertData(
+    expectedColumns: string[],
+    assertionData: Record<string, unknown>,
+    registrationIndex: number,
+    filterContext?: string,
+    validateRowCount?: { condition: boolean; minRowCount: number },
+  ) {
+    const [download] = await Promise.all([
+      this.page.waitForEvent('download'),
+      this.clickProceedToExport(),
+    ]);
+
+    const downloadDir = './downloads';
+    if (!fs.existsSync(downloadDir)) fs.mkdirSync(downloadDir);
+
+    const filePath = path.join(downloadDir, download.suggestedFilename());
+    await download.saveAs(filePath);
+
+    const workbook = XLSX.readFile(filePath);
+    const sheetName = workbook.SheetNames[0];
+    const sheet = workbook.Sheets[sheetName];
+    const data = XLSX.utils.sheet_to_json(sheet, { defval: null }) as Record<
+      string,
+      unknown
+    >[];
+
+    if (data.length === 0) throw new Error('No data found in the sheet');
+
+    if (validateRowCount?.condition) {
+      if (data.length <= validateRowCount.minRowCount) {
+        throw new Error(
+          `Row count validation failed. Expected more than ${validateRowCount.minRowCount} rows, but found ${data.length}.`,
+        );
+      }
+    }
+
+    const rowToAssert = filterContext
+      ? data.find((row) =>
+          Object.values(row).some((value) =>
+            value?.toString().includes(filterContext),
+          ),
+        )
+      : data[registrationIndex];
+
+    if (!rowToAssert) throw new Error('No data found to assert');
+
+    const actualColumns = Object.keys(rowToAssert).map((col) =>
+      col.toLowerCase().trim(),
+    );
+    const normalizedExpectedColumns = expectedColumns.map((col) =>
+      col.toLowerCase().trim(),
+    );
+
+    if (
+      !normalizedExpectedColumns.every((col) => actualColumns.includes(col)) ||
+      normalizedExpectedColumns.length !== actualColumns.length
+    ) {
+      throw new Error('Column validation failed');
+    }
+
+    // Use allXlxsColumnsMapping for key mapping
```

Comment:
```suggestion
```

This comment is describing what is happening a few lines below; But when the variable/type gets renamed, the comment gets out-of-date.

Reply:
Removed it